java/icicle/Pair.java

```diff
@@ -15,4 +17,27 @@ public Pair(A a, B b)
     {
         return new Pair<A,B>(a, b);
     }
+
+    public String toString()
+    {
+        return "(" + _fst.toString() + ", " + _snd.toString() + ")";
+    }
+
+    public boolean equals(Object o)
```

Comment:
http://stackoverflow.com/questions/1130433/how-to-avoid-unchecked-conversion-warning-in-java-if-you-use-legacy-libraries

Yah Java